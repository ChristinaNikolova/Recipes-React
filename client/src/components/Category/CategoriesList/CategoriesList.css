#categories {
    height: auto;
    padding-bottom: 40px;
}

.categories-list {
    display: flex;
    justify-content: center;
    align-items: center;
    gap: 20px;
    flex-wrap: wrap;
}