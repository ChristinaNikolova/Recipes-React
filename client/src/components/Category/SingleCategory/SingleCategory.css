.custom-card-category {
    width: 32%;
    margin-bottom: 15px;
}

.custom-card-category h3.card-header {
    background-color: #54BAB9;
    font-weight: 400;
    letter-spacing: 1px;
    color: #F7ECDE;
    text-align: center;
}

.category-img {
    width: 100%;
    height: 300px;
    object-fit: cover;
    border: 0.1rem hidden;
    padding: 10px;
    background-color: white;
}

.category--recipe-count {
    border-top: 1px solid lightgrey;
    border-bottom: 1px solid lightgrey;
    padding: 18px 15px;
    margin-bottom: unset;
    font-size: 17px;
    letter-spacing: .8px;
}

.custom-card-category .fa-book {
    color: #54BAB9;
}

.card-body .custom-btn {
    width: 50%;
    font-size: 18px;
    margin-bottom: 10px;
}