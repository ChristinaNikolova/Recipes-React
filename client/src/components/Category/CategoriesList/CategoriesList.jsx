import { useState, useEffect } from 'react';

import * as categoriesService from '../../../services/categoriesService.js';
import SingleCategory from '../SingleCategory/SingleCategory.jsx';

import './CategoriesList.css';

function CategoriesList() {
    const [categories, setCategories] = useState([]);

    useEffect(() => {
        categoriesService
            .all()
            .then(categories => setCategories(categories))
            .catch(err => console.error(err));
    }, []);

    return (
        <div className="categories-wrapper">
            <div className="container">
                <div className="col-md-12">
                    <h1 className="text-center pt-2 cursive-font-style">All Categories</h1>
                    <hr />
                </div>
                <div className="row">
                    {categories
                        .map(c => <SingleCategory
                            key={c.id}
                            id={c.id}
                            name={c.name}
                            picture={c.picture}
                            recipesCount={c.recipesCount} />)}
                </div>
            </div>
        </div>
    );
}

export default CategoriesList;