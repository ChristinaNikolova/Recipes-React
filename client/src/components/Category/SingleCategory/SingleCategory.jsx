import { Link } from 'react-router-dom';

import './SingleCategory.css';

function SingleCategory({ id, name, picture, recipesCount }) {
    return (
        <div className="custom-card-category">
            <div className="card">
                <h3 className="card-header cursive-font-style">{name}</h3>
                <img className="category-img" src={picture} alt="category-pic"></img>
                <ul className='category--recipe-count'>
                    <li>
                        <i className="fas fa-book mr-1"></i>
                        Recipes in category: {recipesCount}
                    </li>
                </ul>
                <div className="card-body">
                    <Link to={`/recipes/current-category/${id}`}>
                        <button className="custom-btn">See recipes</button>
                    </Link>
                </div>
            </div>
        </div>
    );
}

export default SingleCategory;