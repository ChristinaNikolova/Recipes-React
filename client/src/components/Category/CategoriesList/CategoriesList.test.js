import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import {act} from "react-dom/test-utils";

import CategoriesList from './CategoriesList';
import SingleCategory from '../SingleCategory/SingleCategory';

describe("CategoriesList", () => {
    let wrapper;
    // let categories;

    beforeEach(() => {
        wrapper = shallow(<CategoriesList />);
    });

    it('should render CategoriesList component', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("should render the count of the categories", async () => {
        await act(async () => {
            await Promise.resolve(wrapper);
            await new Promise(resolve => setImmediate(resolve));
            wrapper.update();
        });
        console.log(wrapper.debug());
    });
});

