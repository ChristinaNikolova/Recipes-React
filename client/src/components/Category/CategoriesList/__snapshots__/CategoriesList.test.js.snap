// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoriesList should render CategoriesList component 1`] = `
<div
  className="categories-wrapper"
>
  <div
    className="container"
  >
    <div
      className="col-md-12"
    >
      <h1
        className="text-center pt-2 cursive-font-style"
      >
        All Categories
      </h1>
      <hr />
    </div>
    <div
      className="row"
    />
  </div>
</div>
`;
